maestro-studio/web/src/InteractPage.tsx

```diff
@@ -70,28 +37,44 @@ const InteractPage = () => {
         <InteractableDevice
           deviceScreen={deviceScreen}
           onHint={setFooterHint}
-          onInspectElement={setInspectedElement}
+          inspectedElement={inspectedElement}
+          onInspectElement={e => setInspectedElementId(e?.id || null)}
         />
       </div>
-      <div className="flex flex-col flex-1 h-full overflow-hidden border-l shadow-xl">
+      <div className="flex flex-col flex-1 h-full overflow-hidden border-l shadow-xl relative">
         <span className="px-6 py-4 font-bold font-mono border-b text-lg cursor-default">$ maestro studio</span>
         <div className="p-6 h-full overflow-hidden">
-          <ReplView onError={setReplError}/>
+          <ReplView
+            input={input}
+            onInput={setInput}
+            onError={setReplError}
+          />
         </div>
         <div
           className="flex items-center gap-1 justify-center px-3 bg-slate-50 border-t h-10 text-slate-500 overflow-hidden whitespace-nowrap data-[error]:bg-red-100 data-[error]:text-red-800"
           data-error={footerHint ? null : replError}
         >
           {footerHint || replError}
         </div>
+        {inspectedElement && (
+          <ActionModal
```

Comment:
Something that I noticed when testing this out -- having to swap between mouse and keyboard when building my flow is a little tedious, so I wonder if we should also consider clicking on an element in action upload as the run action? That removes having to hit enter to confirm each command and makes for a more smooth UX in my opinion.

Reply:
I was thinking about this as well but maybe double-click makes sense? Check out Raycast UX for instance

Reply:
Yeah I like that, double-click sounds great!